question-servers/freeform.js

```diff
@@ -1173,13 +1190,47 @@ module.exports = {
       (cachedData, cacheHit) => {
         // function to process the cachedData, whether we
         // just rendered it or whether it came from cache
-        const { courseIssues, html, renderedElementNames } = cachedData;
-        callback(null, courseIssues, html, renderedElementNames, cacheHit);
+        callback(null, {
+          ...cachedData,
+          cacheHit,
+        });
       },
       callback // error-handling function
     );
   },
 
+  renderPanelInstrumented: async function (
+    panel,
+    pc,
+    submission,
+    variant,
+    question,
+    course,
+    locals,
+    context
+  ) {
+    return instrumented(`freeform.renderPanel:${panel}`, async (span) => {
+      span.setAttributes({
+        panel,
+        'variant.id': variant.id,
+        'question.id': question.id,
+        'course.id': course.id,
+      });
+      /** @type {RenderPanelResult} */
+      const result = await promisify(module.exports.renderPanel)(
+        panel,
+        pc,
+        variant,
+        submission,
+        course,
+        locals,
+        context
+      );
+      span.setAttribute('cache.status', result.cacheHit ? 'hit' : 'miss');
```

Comment:
Cool!

Reply:
Yes! Once we prove this out in prod, I'd like to rip out the cache tracking that we currently have in the `page_view_logs` table.

Reply:
Totally agreed.